### PR TITLE
Move a bit the code to 2018 edition. Fix most of the clippy errors.

### PR DESCRIPTION
**Pull request by aplanas**
I would like to move the codebase into the 2018 edition.

This PR is:

* Dropping some of the `extern crate`
* Fixing some typos in Cargo.toml
* Fixing most of the clippy errors / warnings

Note that there are still two clippy errors related with complexity. I am not confident to address them with my current rust knowledge, but I will try in a future PR (if you do not fix them before).

- [X] cargo fmt has been run
- [X] cargo test has been run and passes
- [ ] design document included (if relevant)

### CODE REVIEW

**Comment by Firstyear** [comment]
This is an awesome change @aplanas, I'm reviewing it now. :) Thanks so much!

File: kanidmd/src/lib/actors/v1_write.rs [hunk 1]
```diff
@@ -765,7 +758,7 @@ impl Handler<PurgeTombstoneEvent> for QueryServerWriteV1 {
 
     fn handle(&mut self, msg: PurgeTombstoneEvent, _: &mut Self::Context) -> Self::Result {
```
Comment (Firstyear) [inline kanidmd/src/lib/actors/v1_write.rs]: It looks like you've removed the result here?

File: kanidmd/src/lib/actors/v1_write.rs [hunk 2]
```diff
@@ -786,7 +778,7 @@ impl Handler<PurgeRecycledEvent> for QueryServerWriteV1 {
 
     fn handle(&mut self, msg: PurgeRecycledEvent, _: &mut Self::Context) -> Self::Result {
         let mut audit = AuditScope::new("purge recycled");
-        let res = audit_segment!(&mut audit, || {
+        audit_segment!(&mut audit, || {
```
Comment (Firstyear) [inline kanidmd/src/lib/actors/v1_write.rs]: And here?

**Comment by Firstyear** [comment]
Beside those two comments about the returns this change is awesome - huge amount of work, really happy with it. You've made a lot of good improvements to the apis and efficiency, so I'll be happy to merge it shortly :D Thanks so much!

File: kanidmd/src/lib/actors/v1_write.rs (hunk 1 above)
Comment (aplanas) [reply kanidmd/src/lib/actors/v1_write.rs]: Good one. Here clippy complains about that we assign to `res` the unit value. `type Result = ()` is defined to this trait, so the only value that can be returned is `()` for `handle` IIUC.

File: kanidmd/src/lib/actors/v1_write.rs (hunk 2 above)
Comment (aplanas) [reply kanidmd/src/lib/actors/v1_write.rs]: Same. Only `()` can be returned.

**Comment by Firstyear** [comment]
Okay, I'll merge and investigate those two functions separately. Thanks so much!

File: kanidmd/src/lib/actors/v1_write.rs (hunk 2 above)
Comment (Firstyear) [reply kanidmd/src/lib/actors/v1_write.rs]: Okay I looked and I can see it now, good spot. Thanks!

**Pull request merged**